src/Channel.hpp

```diff
@@ -36,73 +36,219 @@ class Channel
+  inline float getBitVolts() const { return m_bitVolts; }
 
   /**
-   * @brief Name of the channel.
+   * @brief Get the name of the array group the channel belongs to.
+   * @return The groupName value.
    */
-  std::string name;
+  inline std::string getGroupName() const { return m_groupName; }
 
   /**
-   * @brief Name of the array group the channel belongs to.
+   * @brief Get the name of the channel
+   * @return The name value.
    */
-  std::string groupName;
+  inline std::string getName() const { return m_name; }
 
   /**
-   * @brief Index of array group the channel belongs to.
+   * @brief Get the array group index the channel belongs to
+   * @return The groupIndex value.
    */
-  SizeType groupIndex;
+  inline SizeType getGroupIndex() const { return m_groupIndex; }
 
   /**
-   * @brief Index of channel within the recording array.
+   * @brief Get the index of the channel within the recording array.
+   * @return The localIndex value.
    */
-  SizeType localIndex;
+  inline SizeType getLocalIndex() const { return m_localIndex; }
 
   /**
-   * @brief Index of channel across the recording system.
+   * @brief Get the index of the channel across the recording system.
+   * @return The globalIndex value.
    */
-  SizeType globalIndex;
+  inline SizeType getGlobalIndex() const { return m_globalIndex; }
 
   /**
-   * @brief Coordinates of channel (x, y, z) within the recording array.
+   * @brief Get the coordinates of channel (x, y, z) within the recording array.
+   * @return The position value.
    */
-  std::array<float, 3> position;
+  inline const std::array<float, 3>& getPosition() const { return m_position; }
 
   /**
-   * @brief Comments about the channel.
+   * @brief Get comments about the channel
+   * @return The comments value.
    */
-  std::string comments;
+  inline std::string getComments() const { return m_comments; }
+
+  /**
+   * @brief Set comments about the channel.
+   * @param comments The comments to set.
+   */
+  inline void setComments(const std::string& comments)
+  {
+    m_comments = comments;
+  }
+
+  /**
+   * @brief Set coordinates of channel (x, y, z) within the recording array.
+   * @param position The position to set.
+   */
+  inline void setPosition(const std::array<float, 3>& position)
+  {
+    m_position = position;
+  }
+
+  /**
+   * @brief Set index of channel across the recording system.
+   * @param globalIndex The globalIndex to set.
+   */
+  inline void setGlobalIndex(const SizeType globalIndex)
+  {
+    m_globalIndex = globalIndex;
+  }
+
+  /**
+   * @brief Set index of channel within the recording array.
+   * @param localIndex The localIndex to set.
+   */
+  inline void setLocalIndex(const SizeType localIndex)
+  {
+    m_localIndex = localIndex;
+  }
+
+  /**
+   * @brief Set index of array group the channel belongs to.
+   * @param groupIndex The groupIndex to set.
+   */
+  inline void setGroupIndex(const SizeType groupIndex)
+  {
+    m_groupIndex = groupIndex;
+  }
```

Comment:
I'm not sure if we want the user to be able to modify the indexes or the group information for the channel after the object is instantiated.

Reply:
Sounds reasonable. Please let me know which setters you would like me to remove or feel free to just make edits to the PR directly.

Reply:
Just pushed a change with the setters I think we should remove for now.